optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/constraintweight/descriptor/ConstraintWeightDescriptor.java

```diff
@@ -75,10 +77,14 @@ public MemberAccessor getMemberAccessor() {
     }
 
     public Function<Solution_, Score<?>> createExtractor() {
-        MemberAccessor constraintConfigurationMemberAccessor = constraintConfigurationDescriptor.getSolutionDescriptor()
-                .getConstraintConfigurationMemberAccessor();
+        SolutionDescriptor<Solution_> solutionDescriptor = constraintConfigurationDescriptor.getSolutionDescriptor();
+        MemberAccessor constraintConfigurationMemberAccessor =
+                solutionDescriptor.getConstraintConfigurationMemberAccessor();
         return (Solution_ solution) -> {
-            Object constraintConfiguration = constraintConfigurationMemberAccessor.executeGetter(solution);
+            Object constraintConfiguration =
+                    Objects.requireNonNull(constraintConfigurationMemberAccessor.executeGetter(solution),
+                            "Constraint configuration provider (" + constraintConfigurationMemberAccessor +
+                                    ") is null.");
```

Comment:
```suggestion
                                    ") returns null.");
```

the memberAccessor isn't null, it returns null.